akka-http/src/main/scala/akka/http/scaladsl/server/directives/FileAndResourceDirectives.scala

```diff
@@ -222,7 +222,14 @@ object FileAndResourceDirectives extends FileAndResourceDirectives {
             ResourceFile(url, entry.getSize, entry.getTime)
           }
         } finally jar.close()
-      case _ ⇒ None
+      case _ ⇒
+        val conn = url.openConnection()
```

Comment:
Less likely, yes. But e.g. a `URLClassLoader` for a remote JAR file could still have network activity as a result to requesting a resource, depending on how it's implemented. Of course, I don't know if it would then actually apply these timeouts to its own connections.